invenio_rdm_records/github/release.py

```diff
@@ -0,0 +1,315 @@
+            else None
+        )
+
+        return data
+
+    def resolve_record(self):
+        """Resolves an RDM record from a release."""
+        recid = self._retrieve_record_by_uuid(self.release_object.record_id)
+        return current_rdm_records_service.read(system_identity, recid.pid_value)
+
+    def _is_first_release(self):
+        """Checks whether the current release is the first release of the repository."""
+        return self.repository_object.releases.count() == 0
+
+    def _retrieve_recid_by_uuid(self, rec_uuid):
+        """Retrieves a persistent identifier given its objects uuid.
+
+        Helper function.
+        """
+        recid = PersistentIdentifier.get_by_object(
+            pid_type="recid",
+            object_uuid=rec_uuid,
+            object_type="rec",
+        )
+        return recid
+
+    def publish(self):
+        """Publish GitHub release as record.
+
+        Drafts and records are created using the current records service.
+        The following steps are run inside a single transaction:
+
+        - Create a draft.
+        - The draft's ownership is set to the user's id via its parent.
+        - Upload files to the draft.
+        - Publish the draft.
+
+        In case of failure, the transaction is rolled back and the release status set to 'FAILED'
+
+        :raises ex: any exception generated by the records service (e.g. invalid metadata)
+        """
+        try:
+            self.release_object.status = ReleaseStatus.PROCESSING
+            data = {
+                "metadata": self.metadata,
+                "access": {"record": "public", "files": "public"},
+                "files": {"enabled": True},
+            }
+
+            with UnitOfWork(db.session) as uow:
+                if self._is_first_release():
+                    draft = current_rdm_records_service.create(
+                        self.user_identity, data, uow=uow
+                    )
+                else:
+                    # Retrieve latest record id and its recid
+                    latest_record_uuid = self.repository_object.latest_release(
+                        ReleaseStatus.PUBLISHED
+                    ).record_id
+
+                    recid = self._retrieve_recid_by_uuid(latest_record_uuid)
+
+                    # Create a new version and update its contents
+                    new_version_draft = current_rdm_records_service.new_version(
+                        self.user_identity, recid.pid_value, uow=uow
+                    )
+                    draft = current_rdm_records_service.update_draft(
+                        self.user_identity, new_version_draft.id, data, uow=uow
+                    )
+
+                # Validate the release files are fetchable
+                self._test_zipball()
+
+                # Upload files to draft
+                draft_file_service = current_rdm_records_service.draft_files
+
+                draft_file_service.init_files(
```

Comment:
I know we discuss it, so this is just curiosity. Why not use the FETCH files? need to know when to comit?

Reply:
I discussed it with @slint, iirc we went for the simpler option, for now, to integrate this as soon as possible. 

But if I recall correctly, the authentication part was the major block we thought of. Right now we authenticate the user through the github client and then we request the release file.